packages/flutter/lib/src/cupertino/activity_indicator.dart

```diff
@@ -128,13 +143,16 @@ class _CupertinoActivityIndicatorPainter extends CustomPainter {
     canvas.save();
     canvas.translate(size.width / 2.0, size.height / 2.0);
 
+    // Rotate so that the top tick is the first to be drawn.
+    canvas.rotate(math.pi / 2);
+
     final int activeTick = (_kTickCount * position.value).floor();
 
-    for (int i = 0; i < _kTickCount; ++ i) {
+    for (int i = ((_kTickCount - 1) * progress).toInt(); i >= 0; --i) {
       final int t = (i + activeTick) % _kTickCount;
-      paint.color = activeColor.withAlpha(_alphaValues[t]);
+      paint.color = activeColor.withAlpha(progress < 1 ? _alphaValues[0] : _alphaValues[t]);
       canvas.drawRRect(tickFundamentalRRect, paint);
-      canvas.rotate(-_kTwoPI / _kTickCount);
+      canvas.rotate(_kTwoPI / _kTickCount);
```

Comment:
If you change the rotate direction here, doesn't it counteract the iteration direction change?

Reply:
There's three things all going on here at once: 1) looping through rendering each tick when spinning, 2) pulling out an opacity for each tick, and 3) incrementally revealing each tick when `progress < 1`.

The first two don't really matter what sequence you do them in (or rotate) as long as you get the indexes correct for the colours (this is because in the finish, all 12 of them need to be drawn). However, the last case /must/ reveal the ticks in specific order (clockwise).

Reply:
Thank for cleaning this up! Now the `_alphaValues` array is a lot easier to reason about.